android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerModule.kt

```diff
@@ -446,23 +453,18 @@ class RNGestureHandlerModule(reactContext: ReactApplicationContext?) :
     }
   }
 
-  @ReactMethod(isBlockingSynchronousMethod = true)
-  override fun install(): Boolean {
-    reactApplicationContext.runOnJSQueueThread {
-      try {
-        SoLoader.loadLibrary("gesturehandler")
-        val jsContext = reactApplicationContext.javaScriptContextHolder!!
-        decorateRuntime(jsContext.get())
-      } catch (exception: Exception) {
-        Log.w("[RNGestureHandler]", "Could not install JSI bindings.")
-      }
-    }
+  private external fun initHybrid(): HybridData
+  private external fun getBindingsInstallerCxx(): BindingsInstallerHolder
+  private external fun decorateUIRuntime()
 
+  override fun getBindingsInstaller() = getBindingsInstallerCxx()
+
+  @ReactMethod(isBlockingSynchronousMethod = true)
+  override fun installUIRuntimeBindings(): Boolean {
+    decorateUIRuntime()
```

Comment:
Can't we just inline `decorateUIRuntime` here?

Reply:
I guess we could, but the module is a hybrid class and `decorateUIRuntime` is implemented in C++.

Reply:
Ideally, I'd like to share code for runtime decoration between platforms, but I really don't want to configure shared c++ code in this PR as well (it's already done in the `display: contents` branch) and resolve conflicts related to that. I'd like to merge those PRs into one branch soon-ish, though.

Reply:
Implemented code sharing in this PR. There's no longer a wrapper around the native method.